src/emoji/data.js

```diff
@@ -47,53 +49,40 @@ export const emojiTypeFromReactionType = (reactionType: ReactionType): EmojiType
  */
 export const getFilteredEmojis = (
   query: string,
-  activeImageEmojiByName: $ReadOnly<{| [string]: ImageEmojiType |}>,
-): $ReadOnlyArray<{| emoji_type: EmojiType, name: string, code: string |}> => {
-  // We start by making a map from matching emoji names to a number
-  // representing how good a match it is: 0 for a prefix match, 1 for a
-  // match anywhere else in the string.
-
-  const matchingUnicodeEmoji = objectEntries(unicodeCodeByName)
-    .map(([name, code]) => {
-      // This logic does not do any special handling for things like
-      // skin-tone modifiers or gender modifiers, since Zulip does not
-      // currently support those: https://github.com/zulip/zulip/issues/992.
-      // Once support is added for that, we may want to come back here and
-      // modify this logic, if for instance, there is a default skin-tone
-      // setting in the webapp that we want to also surface here. (or
-      // perhaps it will be best to leave it as is - that's a product
-      // decision that's yet to be made.) For the time being, it seems
-      // better to not show the user anything if they've searched for an
-      // emoji with a modifier than it is to show them the non-modified
-      // emoji, hence the very simple matching.
-      const matchesEmojiLiteral = parseUnicodeEmojiCode(code) === query;
-      const matchesEmojiName = Math.min(1, name.indexOf(query));
-      return [name, matchesEmojiLiteral ? 0 : matchesEmojiName];
-    })
-    .filter(([_, i]) => i !== -1);
+  activeImageEmoji: $ReadOnlyArray<EmojiForShared>,
+): $ReadOnlyArray<EmojiForShared> => {
+  const matcher = typeahead.get_emoji_matcher(query);
+  const allMatchingEmoji: Map<string, EmojiForShared> = new Map();
+  const unicodeEmojiObjects = objectEntries(unicodeCodeByName).map(([name, code]) => ({
+    emoji_type: 'unicode',
+    emoji_name: name,
+    emoji_code: code,
+  }));
```

Comment:
We're now building up new objects here on every call to `getFilteredEmojis`, for each of our 1550 entries for Unicode emoji. That's a regression from the previous revision (at https://github.com/zulip/zulip-mobile/commit/19a48fb157c8c251f4c59db4b356b01fdca3275c).

Reply:
Did you perhaps mis-link the previous revision? That commit is also creating a map from the unicodeByName selector. I guess the solution here is to construct another redux selector.

Reply:
The difference is that that one is doing so at the top level of the module. That means it happens once, when the module is first imported, instead of every time `getFilteredEmojis` is called.

There's no need for a memoizing selector, because the data never changes. `unicodeCodeByName` is a constant data structure, not a selector.

Reply:
Fixed.